### ISSUE TITLE
iOS Support - Persist Cookies on App Pause

### ISSUE THREAD

**Issue by gregberns** [DriveTimeInc/cordova-plugin-cookie-persistence#3]
We'd like to support iOS devices with this plugin but the are technical challenges due to Apple's SDK implementation. This plugin does two things, this issue focuses on persisting the cookies:

1. **Persist cookies to storage when the application is paused** - This issue (#3)
2. Restore persisted cookies to the web view so they're in included in first request - See Issue #4

Prefer to support iOS 10+.

## Basic Intuition (taken from Android implementation)

1. Detect when the application is paused
2. Extract the cookie string from the web view
3. Store the cookie string in a text file

### Detect onPause

There are two ways to to this:
1. Add a native onPause listener which triggers a method to extract the cookies
2. In JS, add an onPause listener which calls get cookies (`document.cookie`), and sends them to the native layer

## Technical Challenges

Having JS call into the native code with cookies to store should work fine, but using all native code may be better. Lets look at options on how to do this.

To extract cookies on pause only through native code, there are known issues with getting access to the cookies reliably. See: 

* [WKWebView Persistent Storage of Cookies](https://stackoverflow.com/questions/39772007/wkwebview-persistent-storage-of-cookies) - There are known issues getting cookies and it seems like people are just shooting in the dark

* [Getting all cookies from WKWebView](https://stackoverflow.com/questions/33156567/getting-all-cookies-from-wkwebview) - the `NSHTTPCookieStorage.sharedHTTPCookieStorage()` method does not appear to work. The `httpCookieStore` method doesn't support iOS 10 which would be good to support at this time.

### Event Listener

```
/**
 * Register the listener for pause and resume events.
 */
- (void) observeLifeCycle
{
    NSNotificationCenter* listener = [NSNotificationCenter
                                      defaultCenter];

    [listener addObserver:self
                 selector:@selector(onPause)
                     name:UIApplicationDidEnterBackgroundNotification
                   object:nil];
}

/**
 * Call on app pause
 */
- (void) onPause
{
    NSLog(@"CookiePersistencePlugin - onPause");

    //fetch cookies from js layer

    //storeCookies
    //[self storeCookies:startPageUrl];
}
```

**Comment by gregberns** [comment]
Fix Proposed in PR #5
It uses an onPause listener in the JS layer to get the cookies with `document.cookies`, then passes the cookie string to the native layer to be persisted.

**Comment by kyleamadio** [comment]
If you can expose a method to call the storage logic from JS at anytime - we can then call the code to refresh the Cookies details whenever we make a HTTPs call to the server. That way we will not need to worry about Exit/Closing of the app as the developer can manage the cookie refresh as they are loaded from the the web server.

**Comment by gregberns** [comment]
Good idea. Since that request applies to both iOS and Android I'll add it as a new issue: #6

**Comment by gregberns** [comment]
Should work now after #8
